src/transform/src/join_implementation.rs

```diff
@@ -100,12 +100,29 @@ impl JoinImplementation {
             for index in 0..inputs.len() {
                 // We can work around filters, as we can lift the predicates into the join execution.
                 let mut input = &mut inputs[index];
-                while let RelationExpr::Filter {
-                    input: inner,
-                    predicates: _,
-                } = input
-                {
-                    input = inner;
+                // while let RelationExpr::Filter {
+                //     input: inner,
+                //     predicates: _,
+                // } = input
+                // {
+                //     input = inner;
+                // }
+                loop {
+                    match input {
+                        RelationExpr::Filter {
+                            input: inner,
+                            predicates: _,
+                        } => {
+                            input = inner;
+                        }
+                        RelationExpr::Project {
+                            input: inner,
+                            outputs: _,
+                        } => {
+                            input = inner;
+                        }
```

Comment:
This looks like it has the potential to mess with the following logic that names the columns on which arrangements are keyed. For filters it was nbd because they field layout isn't changed, but for projections it may be. Additionally, some arrangements may be non-viable because their columns are projected away.

Reply:
Oops, yeah, I meant to change this 😬